Sources/SwiftSyntax/Trivia.swift

```diff
@@ -42,6 +42,101 @@ public struct Trivia: Sendable {
+        switch piece {
+        case .lineComment(let text):
+          return text.hasPrefix("// ")
+        case .docLineComment(let text):
+          return text.hasPrefix("/// ")
+        default:
+          return true
+        }
+      }
+    }()
+
+    // Returns a substring with leading and trailing spaces removed.
+    func trimWhitespace(_ text: Substring) -> Substring {
+        let trimmed = text.drop(while: { $0 == " " })
+        let reversed = trimmed.reversed()
+        let trimmedEnd = reversed.drop(while: { $0 == " " })
+        let final = trimmedEnd.reversed()
+        return Substring(final)
+    }
+
+    // Strips /* */ markers and aligns content by removing common indentation.
+    func processBlockComment(_ text: Substring) -> String {
+      var lines = text.split(separator: "\n", omittingEmptySubsequences: false)
+
+      let minIndentation =
+        lines
+        .dropFirst()
+        .filter { !$0.isEmpty }
+        .map { $0.prefix { $0 == " " }.count }
+        .min() ?? 0
+
+      var firstLineRemoved = false;
+      var firstLine = lines[0]
+      if trimWhitespace(firstLine) == "/*" || trimWhitespace(firstLine) == "/**" {
+          lines.removeFirst()
+          firstLineRemoved = true;
+      } else {
+          firstLine = firstLine.hasPrefix("/**") ? firstLine.dropFirst(3) : firstLine.dropFirst(2)
+          while firstLine.first?.isWhitespace == true {
+              firstLine = firstLine.dropFirst()
+          }
+          lines[0] = firstLine
+      }
+      
+      if let lastLine = lines.last {
+          if trimWhitespace(lastLine) == "*/" {
+              lines.removeLast()
+          } else {
+              var lastLine = lines[lines.count - 1]
+              lastLine = lastLine.hasSuffix("*/") ? lastLine.dropLast(2) : lastLine
+              while lastLine.last?.isWhitespace == true {
+                  lastLine = lastLine.dropLast()
+              }
+              lines[lines.count - 1] = lastLine
+          }
+      }
+
+      let unindentedLines = lines.enumerated().map { index, line -> Substring in
+        if index == 0 && firstLineRemoved == false {
+            return line
+        }
+        return line.count >= minIndentation ? line.dropFirst(minIndentation) : line
+      }
+
+      return unindentedLines.joined(separator: "\n")
+    }
+
+    for piece in pieces {
+      switch piece {
+      case .blockComment(let text), .docBlockComment(let text):
+        let processedText = processBlockComment(text[...])
+        comments.append(processedText[...])
+
+      case .lineComment(let text):
+        let prefix = allLineCommentsHaveSpace ? "// " : "//"
+        comments.append(text.dropFirst(prefix.count))
+
+      case .docLineComment(let text):
+        let prefix = allLineCommentsHaveSpace ? "/// " : "///"
+        comments.append(text.dropFirst(prefix.count))
+
+      default:
+        break
+      }
+    }
+    return comments.joined(separator: "\n")
```

Comment:
I just realized that if you have `/* abc */ /* def */` as your trivia, then this will return 

```
abc
def
```
as the comment value (concatenated by newline), which isn’t what I would expect. I’m leaning towards making `commentValue` return an Optional and return `nil` if:
- No comment is in the trivia
- The trivia contains more than one block comment or a block comment and a line comment.